pkg/connection/manager.go

```diff
@@ -65,10 +66,11 @@ type DetailsManager struct {
 	client       client.Client
 	newConfig    func() StoreConfig
 	storeBuilder StoreBuilderFn
+	tlsConfig    *tls.Config
 }
 
 // NewDetailsManager returns a new connection DetailsManager.
-func NewDetailsManager(c client.Client, of schema.GroupVersionKind, o ...DetailsManagerOption) *DetailsManager {
+func NewDetailsManager(c client.Client, of schema.GroupVersionKind, tlsConfig *tls.Config, o ...DetailsManagerOption) *DetailsManager {
```

Comment:
What do you think about introducing the `tlsConfig` as an option? 

- Not all secret stores need it, e.g. `Kubernetes` 
- Not to have a breaking change in the function signature. 


```
func WithTLSConfig(tlsConfig *tls.Config) DetailsManagerOption {
	return func(m *DetailsManager) {
		m.tlsConfig = tlsConfig
	}
}
```